test/reporter_output_test.cc

```diff
@@ -52,6 +52,46 @@ ADD_CASES(TC_JSONOut, {{"\"name\": \"BM_error\",$"},
 ADD_CASES(TC_CSVOut, {{"^\"BM_error\",,,,,,,,true,\"message\"$"}});
 
 // ========================================================================= //
+// ------------------------ Testing No Arg Name Output -----------------------
+// //
+// ========================================================================= //
+
+void BM_no_arg_name(benchmark::State& state) {
+  while (state.KeepRunning()) {
+  }
+}
+BENCHMARK(BM_no_arg_name)->Arg(3);
+ADD_CASES(TC_ConsoleOut, {{"^BM_no_arg_name/3 %console_report$"}});
+ADD_CASES(TC_JSONOut, {{"\"name\": \"BM_no_arg_name/3\",$"}});
+ADD_CASES(TC_CSVOut, {{"^\"BM_no_arg_name/3\",%csv_report$"}});
+
+// ========================================================================= //
+// ------------------------ Testing Arg Name Output ----------------------- //
+// ========================================================================= //
+
+void BM_arg_name(benchmark::State& state) {
+  while (state.KeepRunning()) {
+  }
+}
+BENCHMARK(BM_arg_name)->Arg(3)->ArgName("first");
```

Comment:
Well, to preserve current behaviour, the output will be exactly as in the current version if you do not specify `ArgName`. If you have more arguments than names, than only the first will be named.
One can find it a possible source of incoherence, but that's similar to the behaviour of `Args`, imagine calling it as `Args({1, 2, 3})->Args({1, 2})->Args({1,2,3,4,5})` - that's user's responsibility to define all the arguments its benchmark will need.

For your suggestion, I find it a bit "heavy" to set argument names each time, I would certainly not want to have to write this: `Args{{"first", 1}, {"second", 3}}->Args{{"first", 2}, {"second", 5}}` and so on. And it is disturbing that the same argument can be named differently each time (or isn't it?)

Anyway, your remarks made me think of a case that I didn't anticipate: the user might want to name only few arguments, e.g. `ArgNames({"first", "", "third"})`, my current PR will write `BM_benchmark/first:1/:2/third:3` (note the semicolon before 2). I'll correct it ASAP.